docs/guides/computed-properties.mdx

```diff
@@ -49,14 +113,20 @@ const memoizedDoubled = memoize((snap) => snap.count * 2)
 const state = proxy({
   count: 1,
   text: 'hello',
-  get doubled(): { return memoizedDoubled(snapshot(state)) },
+  get doubled() {
+    return memoizedDoubled(snapshot(state))
+  },
 })
 ```
 
-In this case, even if the `text` property changes, the memoized function won't be re-executed.
+With this implementation, when `text` property changes (but `count` has not), the memoized function won't be re-executed.
 
 ## Note about using `this`
 
-You can use `this` instead of `state` in object getters and setters.
-You should be familiar with how JS `this` works,
-as `this` can point to proxy state or snapshot object depending on usage.
+You can use `this` inside of getters and setters, however you should be familiar with how JS `this` works: basically that `this` is whatever object you invoked the call against.
+
+So if you call `state.doubled`, then `this` will be the `state` proxy.
+
+And if you call `snap.doubled`, then `this` will be the snapshot object (except for object getters & setters, where the value is literally copied during the snapshot process, so object getters & setters are never invoked on snapshots, and for them `this` is _always_ the proxy).
```

Comment:
Fwiw @dai-shi I don't meant to just re-start the debate here, but writing this made me realize that if class getters were cached like object getters, it would let us say that `this` is _always_ the proxy object.

(Because snapshots would never have/invoke getters/setters, as the getter is invoked on the proxy during the snapshot process, and then is just a dumb value in the snapshot.)

Which, dunno, seems like a pretty big simplification for users, and might let us reduce the warnings / "advanced usage only" disclaimers that are currently in the docs?

Reply:
just a minor comment:

> let us say that this is always the proxy object.

it would for getters and setters, but not methods in general.